server/controller/dailySchedule.js

```diff
@@ -0,0 +1,82 @@
+import { check, validationResult } from "express-validator";
+// MODELS
+import DailySchedule from "../models/DailySchedule.js";
+import Chore from "../models/Chore.js";
+import User from "../models/User.js";
+
+/*
+ ** This function creates a new schedule for the day
+ ** @param {Object} req - The request object
+ ** @param {Object} res - The response object
+ */
+ export const createDailySchedule = async (req, res) => {
+
+    // Sanitize data
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        return res.status(400).json({ errors: errors.array() });
+    }
+
+    try {
+        // const timeframe = req.querry.timeframe; //extracting timeframe from request
+
+        var dailySchedule = []; // will add chores for the day in here
+
+        // get the user that is requesting the schedule
+        const userID = req.user.id;
+        const user = await User.findOne({ _id: userID }); // search for the user
+        const choreList = user.chores;
+        
+        console.log("Printing chore ids for this user")
+
+        // todays date in ms
+        const today = Date.now();
+
+        for (const choreId of choreList) {
+            // console.log(choreId);
+            const chore = await Chore.findOne({ _id: choreId });
+
+            if (!chore) {
+                return res.status(404).json({ message: "Chore not found" });
+            }
+
+            // mapping interval into days
+            var intervalToDays;
+            if (chore.frequency.interval == "days") {
+                intervalToDays = 1;
+            } else if (chore.frequency.interval == "weeks") {
+                intervalToDays = 7;
+            } else if (chore.frequency.interval == "months") {
+                intervalToDays = 30; // just defaulting to 30
+            } else {
+                intervalToDays = 365; // intterval is in years
+            }
+
+            // calculating how many days this chore is to be repeated
+            const repeatInDays = chore.frequency.quantity * intervalToDays; 
+
+            // calculating difference in days of the chore created vs now
+            const diffInMs = today - chore.createdAt;
+            const diffInDays = Math.floor(diffInMs / (1000 * 60 * 60 * 24));
+
+            // now if the chore is scheduled for today, add it to the schedule list
+            if (diffInDays % repeatInDays == 0) {
+                const scheduleItem = {choreObj: chore, isCompleted: false};
+                dailySchedule.push(scheduleItem);
+            }
+        }
+
+        const newSchedule = new DailySchedule({dailySchedule});
+        // Save to database
+        const savedSchedule = await newSchedule.save();
```

Comment:
Not sure if we actually need to save the schedule to the DB. Perhaps only to speed things up by just saving the schedule for 1-2 days, and then removing it? That way the frontend request can just return the saved schedule if it already exists, else generate it once. And give the frontend the ability to request that the schedule be re-generated. 

I feel like we should only store 3 schedules for each user: yesterday, today, tomorrow. At most, maybe the user could view the next week instead of just the next day, resulting in 9 schedules (yesterday, today, next 7 days). Don't want to really do more than that since it becomes a calendar app. This means we would delete schedules that are more than one day old.

Alternatively, don't store the schedule at all, only store a list of which chores have been checked off (choreId, checked boolean). Generate the schedule using this information and just return it without saving. List of chores that have been checked off would need to be "cleaned" regularly instead (i.e. remove if more than 1 day old). This would be slower since schedule is re-generated each time.